aarch64_qemuvirt/src/main.rs

```diff
@@ -44,33 +36,28 @@ extern "C" fn k_main(_device_tree_ptr: usize) -> ! {
         asm::barrier::dsb(asm::barrier::SY);
     };
 
-    let device_tree_ptr = DTB_ADDR;
-    let device_tree = kernel::device_tree::DeviceTree::new(device_tree_ptr);
-    //
-    let pmm = kernel::mm::PhysicalMemoryManager::from_device_tree(&device_tree, 4096);
-    let mut mm = kernel::mm::MemoryManager::new(pmm);
-    let pagetable = kernel::mm::map_address_space(
-        &device_tree,
-        &mut mm,
-        &[kernel::kernel_console::get_console()],
-    );
-    mm.set_kernel_pagetable(pagetable);
-
-    // ???:
-    // why do we pass the pagetable after into the mm ?
-    //     (can we pass it at initialization)
-    // - also when we map, we reload the page table, enabling the MMU (SCTLR_EL1.M)
-    //   shouldn't that be separate ? ie map just put it in the pagetable, then we can manually reload it / enable paging ?
+    let device_tree = kernel::device_tree::DeviceTree::new(DTB_ADDR);
 
-    kernel::kprintln!("Kernel has been initialized");
+    kernel::globals::PHYSICAL_MEMORY_MANAGER
+        .lock(|pmm| pmm.init_from_device_tree(&device_tree, 4096));
+    kernel::mm::map_address_space(&device_tree, &[&PL011]);
 
-    mm.kernel_map(
-        0x0900_0000,
-        0x0450_0000,
-        kernel::mm::Permissions::READ | kernel::mm::Permissions::WRITE,
+    kernel::kprintln!("PMM has been initialized with the device tree... check");
+    kernel::kprintln!(
+        "kernel's address space has been mapped into the kernel's pagetable... check"
     );
-    let mut uart = Pl011::new(0x0450_0000);
-    uart.write("Uart remaped");
+    kernel::kprintln!("Kernel initialization should be about done.");
+
+    kernel::kprintln!("Testing the remapping capabilities of our pagetable...");
+    kernel::globals::KERNEL_PAGETABLE.lock(|pagetable| {
+        pagetable.map(
+            0x0900_0000.into(),
+            0x0450_0000.into(),
+            kernel::mm::Permissions::READ | kernel::mm::Permissions::WRITE,
+        );
+    });
```

Comment:
This should be done with the test infrastructure you just deleted

Reply:
Euh I guess so ? (This code has always been here).
The tests have been gone since before this PR/branch was created. I don't plan on putting them back in now

Reply:
It was gone before? My bad